src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,142 @@
+package duke.storage;
+
+import duke.task.*;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.nio.file.Path;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import java.time.LocalDate;
+
+/**
+ * Storage class is used to interact with the File System and gather the save data from the last use session.
+ */
+public class Storage {
+
+    private String currentDir = System.getProperty("user.dir");
+    private Path dataDir = Paths.get(currentDir, "src", "main", "data");
+    private Path fileDir = Paths.get(currentDir, "src", "main", "data", "list.txt");
+
+    private static final Pattern pattern = Pattern.compile("^\\[(.)]\\s\\[(.)]\\s(.*?)(?:\\s/..\\s(.*))?$");
+
+    /**
+     * Checks if the /data directory exists.
+     * @return True if it exists, False otherwise.
+     */
+    private boolean dirExists() {
```

Comment:
Will it be better to give a more descriptive name to the method, such as the name is directory present?